src/jobflow_remote/jobs/jobcontroller.py

```diff
@@ -3594,3 +3763,72 @@ def get_flow_leafs(job_docs: list[dict]) -> list[dict]:
                 d.pop(parent_id, None)
 
     return list(d.values())
+
+
+class Upgrader:
+    """
+    Class used to upgrade the database.
+    """
+
+    def __init__(self, job_controller: JobController):
+        self.job_controller = job_controller
+        self.upgrade_functions = {
+            "0.1.3": self._upgrade_to_0_1_3,
+            # "0.1.4": self._upgrade_0_1_3_to_0_1_4,
+        }
+        self._versions = [
+            "before_0_1_3",
+            *sorted(self.upgrade_functions.keys(), key=Version),
+        ]
+
+    def upgrade(self, previous_version, this_version):
+        """Upgrade the database to the new version of Jobflow Remote.
+
+        Parameters
+        ----------
+        previous_version : str
+            Previous version of Jobflow Remote.
+        this_version : str
+            New version of Jobflow Remote.
+        """
+        # TODO: do this with a transaction ?
+        logger.info(
+            f"Starting upgrade of the database from version {previous_version} to version {this_version}"
+        )
+        for prev, new in zip(
+            self._versions, self._versions[1 : self._versions.index(this_version) + 1]
+        ):
+            logger.info(
+                f"Performing incremental upgrade of the database from version {prev} to version {new}"
+            )
+            self.upgrade_functions[new]()
+        _update_version_information(self.job_controller.auxiliary)
+        logger.info(
+            f"Finished upgrade of the database from version {previous_version} to version {this_version}"
+        )
+        return True
+
+    def _upgrade_to_0_1_3(self):
+        self.job_controller.auxiliary.find_one_and_update(
+            filter={"running_runner": {"$exists": True}},
+            update={"$set": {"running_runner": None}},
```

Comment:
Correct, but the upgrade is supposed to be done in the cli and the cli checks (locally, maybe it could/should check in the db also ?) if it is running before doing the upgrade (in the same way a reset could be done in python while a runner is running for example).

Reply:
ok, fine with me